### PR TITLE
Issue 84: Channel not shutdown correctly in TurbineHeatSensor sample

### LINKED ISSUE #84: Channel not shutdown correctly in turbineSensor app
I see the following error when running the standalone turbineSensor app:

```
SEVERE: *~*~*~ Channel io.grpc.internal.ManagedChannelImpl-27 for target tcp://pravega-controller.pravega.svc.cluster.local:9090 was not shutdown properly!!! ~*~*~*
    Make sure to call shutdown()/shutdownNow() and awaitTermination().
java.lang.RuntimeException: ManagedChannel allocation site
        at io.grpc.internal.ManagedChannelImpl$ManagedChannelReference.<init>(ManagedChannelImpl.java:1007)
        at io.grpc.internal.ManagedChannelImpl.<init>(ManagedChannelImpl.java:430)
        at io.grpc.internal.AbstractManagedChannelImplBuilder.build(AbstractManagedChannelImplBuilder.java:337)
        at io.pravega.client.stream.impl.ControllerImpl.<init>(ControllerImpl.java:164)
        at io.pravega.client.stream.impl.ControllerImpl.<init>(ControllerImpl.java:123)
        at io.pravega.client.ClientFactory.withScope(ClientFactory.java:78)
        at io.pravega.client.ClientFactory.withScope(ClientFactory.java:65)
        at io.pravega.example.iot.TurbineHeatSensor.main(TurbineHeatSensor.java:119)
```

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**
Minor fix in `TurbineHeatSensor.java` regarding the use of client factories while creating writers. Note that this PR specifically targets to avoid `java.lang.RuntimeException: ManagedChannel allocation site` exception to be thrown (e.g., the app may throw other exceptions that will be addressed separately).  

**Purpose of the change**
Fixes #84.

**What the code does**
The constructor of `TemperatureSensors` receives by parameter a client factory to create event writers. However, it was creating writers via a global static client factory, which was in turn shared to create readers as well. Creating readers/writers carelessly, jointly with the fact that resources are not correctly closed, could be the cause for the closing channel exception to be thrown. In this fix, we replaced the use of the global static client factory by the one passed by parameter to create writers. While this was enough to avoid the exception to occur, we also decided to close both client factories and writers for the sake of correctness.

**How to verify it**
Execute the `turbineHeatSensor` application and verify that the exception mentioned in the issue does not appear in the console.

### CODE REVIEW

File: scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java [hunk 1]
```diff
@@ -142,6 +142,7 @@ public static void main(String[] args) throws Exception {
         if ( !onlyWrite ) {
             consumeStats.printTotal();
         }
+        clientFactory.close();
 //        ZipKinTracer.getTracer().close();
```
Comment (fpj) [inline scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: Remove this commented line in a new PR, please.

Comment (RaulGracia) [reply scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: Ok, I will do that.

File: scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java [hunk 2]
```diff
@@ -142,6 +142,7 @@ public static void main(String[] args) throws Exception {
         if ( !onlyWrite ) {
             consumeStats.printTotal();
         }
+        clientFactory.close();
```
Comment (fpj) [inline scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: This is a bit confusing as we are closing in the `main` method and in the `runLoop`. Is the contract that the runnable is supposed to close the factory or is this something that needs to be done by the caller? Let's be consistent about how we are closing it.

Comment (RaulGracia) [reply scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: To follow a consistent pattern without changing much the current code: i) now there is only one client factory that is created and closed within the main method. ii) Writers and readers take as input that factory to instantiate an EventStream[Writer/Reader], iii) writers and readers are in charge of closing their own EventStream[Writer/Reader]. In any case, a future PR for issue #119 will be devoted to improve the code of this application, so we can find better ways of managing resources (out of the scope of this PR).

File: scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java [hunk 3]
```diff
@@ -108,28 +109,24 @@ public static void main(String[] args) throws Exception {
...
             consumeStats = new PerfStats(producerCount * eventsPerSec * runtimeSec, reportingInterval, messageSize);
-            SensorReader reader = new SensorReader(producerCount * eventsPerSec * runtimeSec);
+            SensorReader reader = new SensorReader(producerCount * eventsPerSec * runtimeSec, clientFactory);
             executor.execute(reader);
         }
         /* Create producerCount number of threads to simulate sensors. */
         Instant startEventTime = Instant.EPOCH.plus(8, ChronoUnit.HOURS); // sunrise
         for (int i = 0; i < producerCount; i++) {
-            URI controllerUri = new URI(TurbineHeatSensor.controllerUri);
-            ClientFactory factory = ClientFactory.withScope(scopeName, controllerUri);
```
Comment (vijikarthi) [inline scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: Is `ClientFactory` thread safe?

Comment (RaulGracia) [reply scenarios/turbine-heat-sensor/src/main/java/io/pravega/turbineheatsensor/TurbineHeatSensor.java]: The `ClientFactory` object is only used in the constructor of `SensorReader` and `TemperatureSensors` to create writers/readers, and they do not hold any reference to the factory object. This yields that only the main thread is actually operating on the `ClientFactory` (no concurrency), while the multiple running threads operate each one on its own EventStreamReader/Writer.

**Pull request merged**